app/src/main/java/com/android/calendar/Event.java

```diff
@@ -376,6 +399,56 @@ private static Event generateEventFromCursor(Cursor cEvents, Context context) {
         return e;
     }
 
+    /** Android's RRULE code is broken in a way the creates additional events in certain
+     *  circumstances (though never doesn't create the actaul event) so let's use another RRULE
+     *  parser to validate if the event is real or not.
+     *
+     *  In this case we're using lib-recur from https://github.com/dmfs/lib-recur through maven.
+     *
+     **/
+    static int checkRRuleEventDate( String rrule, long startTime, int endDay) {
+        // Convert the startTime into some useable Day/Month/Year values.
+        Date date = new java.util.Date(startTime);
+
+        // We'll use SimpleDateFormat to get the D/M/Y but we also need to set the timezone.
+        SimpleDateFormat sdf = new java.text.SimpleDateFormat();
+        sdf.setTimeZone(java.util.TimeZone.getTimeZone("GMT"));
```

Comment:
is GMT always correct?

Reply:
We're comparing against ```startTime```, which appears to always be UTC/GMT, then displayed/converted to the appropriate timezone later.

Reply:
makes sense, thx!